Source/Src_nd/make_S.F90

```diff
@@ -175,9 +177,11 @@ subroutine make_S_cc_sphr(lev, lo, hi, &
     double precision sigma, xi_term, pres_term, Ut_dot_er
     double precision gradp0(1,0:nr_fine-1)
 
-    double precision, pointer ::       p0_cart(:,:,:,:)
-    double precision, pointer ::   gradp0_cart(:,:,:,:)
-    double precision, pointer ::gamma1bar_cart(:,:,:,:)
+    double precision, allocatable ::       p0_cart(:,:,:,:)
```

Comment:
These types of allocatables in device code are very painful in performance terms. The approach we took in Castro is to allocate them as local Fabs in the C++ loop (so we can take advantage of the memory pool there, and also so that it's not every single thread allocating the data), and then pass in the pointers to the Fortran function. I recommend that you do the same here.